src/LatteContext/LatteContextHelper.php

```diff
@@ -44,4 +52,61 @@ public static function variablesFromTemplateType(string $class): array
         $classType = (new ObjectType($class))->toArray();
         return self::variablesFromType($classType);
     }
+
+    /**
+     * @return Variable[]
+     */
+    public static function variablesFromExpr(?Expr $expr, Scope $scope): array
+    {
+        if ($expr === null) {
+            return [];
+        }
+
+        return LatteContextHelper::variablesFromType($scope->getType($expr));
+    }
+
+    /**
+     * @param class-string|class-string[] $classes
+     */
+    public static function isClass(Node $node, Scope $scope, $classes): bool
```

Comment:
I'm not sure if this is only Latte context thing. Maybe it should be part of resolvers (like we have TemplateTypeResolver), or even better use some built-in function from phpstan?

Reply:
Cirrently it is only latte context thing but conceptually it is not. We could either create some general helper for types in root namespace or create NodeClassResolver.

Reply:
I think there is no such finction in phostan because this helper encapsulates some repated but specific logic.